packages/wrangler/src/pipelines/index.ts

```diff
@@ -65,11 +65,21 @@ async function authorizeR2Bucket(
 	!__testSkipDelaysFlag &&
 		(await retryOnAPIFailure(
 			async () => {
-				await r2.send(
-					new HeadBucketCommand({
-						Bucket: bucketName,
-					})
-				);
+				try {
+					await r2.send(
+						new HeadBucketCommand({
+							Bucket: bucketName,
+						})
+					);
+				} catch (err) {
+					if (err instanceof Error && err.name === "401") {
```

Comment:
Confirmed that these errors are now retried when running `wrangler pipelines create`
<img width="624" alt="Screenshot 2025-01-24 at 8 37 12 AM" src="https://github.com/user-attachments/assets/0feec721-2042-449a-9b7e-59a6f7391174" />